blog/2022-11-01-cfd-protocol-explained.md

```diff
@@ -0,0 +1,229 @@
+Instead, a `commitment_transaction` is introduced, which spends the `lock_output` into the `dlc_output`.
+
+The `dlc_output`, now part of a different transaction in the protocol, is still spent in the same way described previously: a set of CETs define how the `dlc_output` is split up based on the price of BTC in USD.
+The purpose of the `commitment_transaction` is to enforce that the `dlc_output` can only be spent by the CETs which were constructed together with it.
+
+Importantly, the `refund_transaction` works the same as before: it spends the `dlc_output`, which is now an output to the `commitment_transaction`.
+We focus on the CETs for clarity, but everything applies to the `refund_transaction` as was discussed in previous sections.
+
+As an example, we can imagine maker and taker agreeing to roll over the CFD because they're both optimistic that the price movement of BTC will benefit them if they wait longer.
+During `contract_setup`, they created and signed `commitment_transaction_0` and a whole set of CETs along with it.
+During `rollover`, they produce a new `commitment_transaction_1` and a new collection of CETs based on a new `announcement` from `olivia`.
+Since they have the means to publish fully signed `commitment_transaction`s, they can eventually publish `commitment_transaction_1` and settle the CFD based on a newer `attestation` from `olivia`.
+
+##### Punish transaction
+
+But, what happens if either party decides to publish an old `commitment_transaction_0` and settle the CFD based on an old BTC price attested by the oracle?
+That would defeat the purpose of rolling over the CFD.
+This is where the revocation mechanism comes in.
+
+A `commitment_transaction`'s `dlc_output` can actually be spent in one more way.
+In full, the `commitment_transaction` can be spent with:
+
+    1. A series of CETs unlocked by one of `olivia`'s `attestation`s.
+    2. A `refund_transaction` that can only be published at least some time after `olivia` has published the relevant `attestation`.
+    3. One `punishment_transaction` per party, which can be constructed unilaterally if one's counterparty publishes a revoked `commitment_transaction`.
+    This transaction allows the aggrieved party.
+
+During `contract_setup` and every subsequent `rollover`, parties work together to add another spending condition to the `dlc_output`.
+They each generate a revocation keypair `(r, R)` and a publication keypair `(p, P)` and share the public keys with each other.
+The `dlc_output` is no longer a simple 2-of-2 multisig on their public keys `A` and `B`.
+It also includes:
+
+    1. A way for the maker to spend it with knowledge of `a`, `r_maker` and `p_maker`.
+    1. A way for the taker to spend it with knowledge of `b`, `r_taker` and `p_taker`.
+
+Instead of exchanging regular signatures on the `commitment_transactions` with each other, each constructs an adaptor signature which when encrypted and published would leak the other party's publishing secret key.
```

Comment:
```suggestion
Instead of exchanging regular signatures on the `commitment_transactions` with each other, each constructs an adaptor signature which when decrypted and published would leak the other party's publishing secret key.
```